### PR TITLE
Handle unmapped subject buffer during backspace

### PR DESCRIPTION
**Pull request by rchande**
In Razor/Venus the subject buffer may become unmapped. Update the
backspace handling in completion to deal with cases where the subject
buffer is unapped and no CompletionService is available.

### CODE REVIEW

**Comment by rchande** [comment]
Tag @dotnet/roslyn-ide  for review

**Comment by jasonmalinowski** [comment]
Tests?

**Comment by CyrusNajmabadi** [comment]
So, i can see that we use GetCompletionService in a bunch of places int he completion code.  Should we make the other places resilient as well?

**Comment by rchande** [comment]
Good point @CyrusNajmabadi. Added more checks/assertions.
Re @jasonmalinowski's point, do we have any unit test level tests of the Controller any more? If not, this is really difficult to add tests for.

**Comment by mattwar** [comment]
@dotnet-bot test windows_eta_open_prtest please

**Comment by rchande** [comment]
retest windows_release_unit32_prtest please

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_FilterModel.cs [hunk 1]
```diff
@@ -210,7 +210,9 @@ internal partial class Session
                 // matches the text typed so far.
 
                 // Ask the language to determine which of the *matched* items it wants to select.
+                // We have a Document so GetCompletionService should always succeed.
                 var service = this.Controller.GetCompletionService();
+                Contract.ThrowIfNull(service, nameof(service));
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_FilterModel.cs]: This is completely unsafe.  We're calling on the background, and the stuff may be unmapped on the FG.

We should either capture hte service initially and pass it through, or we have to be resilient to it not being aailable here.

Comment (rchande) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session_FilterModel.cs]: Fixed

**Pull request merged**